Sources/Purchasing/StoreKitAbstractions/SK2AppTransaction.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SK2AppTransaction.swift
+//
+//  Created by MarkVillacampa on 26/10/23.
+
+import StoreKit
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+internal struct SK2AppTransaction {
+
+    @available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, visionOS 1.0, *)
+    init(appTransaction: AppTransaction) {
+        self.bundleId = appTransaction.bundleID
+        self.originalApplicationVersion = appTransaction.originalAppVersion
+        self.originalPurchaseDate = appTransaction.originalPurchaseDate
+        self.environment = StoreEnvironment.init(rawValue: appTransaction.environment.rawValue.lowercased())
```

Comment:
Instead of putting the logic to convert here I'd add a constructor in `StoreEnvironment` that takes an `AppStore.Environment` type. Then in there we should log if the type is unrecognized too.

And I'd strongly recommend adding tests for that because this is implicitly assuming that our raw values match StoreKit's.

Reply:
Added constructor and now using exhaustive switch instead of converting from raw values